src/backend/TrafficCourts/Citizen.Service/Features/Tickets/Analyse.cs

```diff
@@ -0,0 +1,117 @@
+
+            public Handler(ILogger<Handler> logger, IOptions<FormRecognizerConfigurationOptions> options)
+            {
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _apiKey = String.IsNullOrEmpty(options.Value.ApiKey) ? throw new ArgumentNullException(nameof(options.Value.ApiKey)) : options.Value.ApiKey;
+                _endpoint = String.IsNullOrEmpty(options.Value.Endpoint) ? throw new ArgumentNullException(nameof(options.Value.Endpoint)) : options.Value.Endpoint;
+            }
+
+            public async Task<AnalyseResponse> Handle(AnalyseRequest request, CancellationToken cancellationToken)
+            {
+                _logger.LogDebug($"Analysing {request.image.FileName}");
+
+                AzureKeyCredential credential = new AzureKeyCredential(_apiKey);
+                DocumentAnalysisClient client = new DocumentAnalysisClient(new Uri(_endpoint), credential);
+
+                using Stream stream = getImageStream(request.image);
+                AnalyzeDocumentOperation analyseDocumentOperation = await client.StartAnalyzeDocumentAsync("ViolationTicket", stream);
+                await analyseDocumentOperation.WaitForCompletionAsync();
+
+                AnalyzeResult result = analyseDocumentOperation.Value;
+
+                // For some reason the Azure.AI.FormRecognizer.DocumentAnalysis.BoundingBoxes are not serialized (always null in JSON)
+                // return new Response(result.Documents[0].Fields);
+
+                // Return a custom mapping of DocumentFields to a structured object for validation and serialization.
+                ViolationTicket violationTicket = map(result);
+                AnalyseResponse response = new AnalyseResponse(violationTicket);
+                return response;
+            }
+
+            private Stream getImageStream(IFormFile image)
+            {
+                // Work around for a "System.ObjectDisposedException : Cannot access a closed file." error
+                // - extract data from the file attachment to an in-memory byte[]
+                // - create a new MemoryStream from the byte[]
+                byte[] fileBytes;
+                using (var fileStream = image.OpenReadStream())
+                using (var ms = new MemoryStream())
+                {
+                    fileStream.CopyTo(ms);
+                    fileBytes = ms.ToArray();
+                }
+                MemoryStream stream = new MemoryStream(fileBytes);
+                return stream;
+            }
+
+            private ViolationTicket map(AnalyzeResult result)
+            {
+                // Iterate over results, creating a structured document
+                ViolationTicket violationTicket = new ViolationTicket();
+                foreach (AnalyzedDocument document in result.Documents)
+                {
+                    violationTicket.confidence = document.Confidence;
+
+                    if (document.Fields != null)

```

Comment:
in C# 7 (C# language spec vs .net framework), the preferred way is to use the `if (value is null)` instead of `if (value == null)`